### ISSUE TITLE
Formalize what modules of `aiida-core` should be importable

### ISSUE THREAD

**Issue by sphuber** [aiidateam/aiida_core#2521]
We should formalize what modules of `aiida-core` should be importable by users. The current suggestion is to expose everything that should be usable by users on the second level of modules. This means that anything a user is supposed to import should be importable from the second nesting, i.e. `from aiida.orm import Node`. With this convention, we guarantee that anything that can be imported like this will not be removed or changed without a proper deprecation pathway. Of course a user can always venture deeper in the module hierarchy but at that point all bets are off with regards to API consistency.

**Comment by sphuber** [comment]
The following is the current suggestion of what classes and functions to expose for each second level module:
```
[aiida.backends]

[aiida.calculations]

[aiida.cmdline]
params.arguments
params.options
params.parameters
params.types
utils.decorators
utils.echo

[aiida.common]
constants?
datastructures
exceptions
extendeddicts
json
links

[aiida.engine]
daemon.runner?
processes.workchains.WorkChain
processes.calcjobs.CalcJob
processes.functions.calcfunction
processes.functions.workfunction
processes.exit_code.ExitCode
launch.run
launch.run_get_node
launch.run_get_pid
launch.submit

[aiida.manage]
backup? (ready?)
manager.get_manager?
fixtures? (ready?)

[aiida.orm]
Node
Data
ProcessNode
CalcFunctionNode
CalcJobNode
WorkFunctionNode
WorkChainNode
ArrayData
BandsData
KpointsData
ProjectionData
TrajectoryData
XyData
Bool
Float
Int
Str
List
ParameterData
CifData
Code
Error?
FolderData
FrozenDict
OrbitalData
RemoteData
SinglefileData
StructureData
UpfData
Comment
Computer
Group
Log
QueryBuilder
User??
loaders: load_node, load_code, load_computer, load_group ...

[aiida.parsers]
Parser

[aiida.plugins]
entry_point
factories

[aiida.restapi]

[aiida.scheduler]
Scheduler

[aiida.sphinxext]

[aiida.tools]
CalculationTools
get_kpoints_path
get_explicit_kpoints_path
structure_to_spglib_tuple
spglib_tuple_to_structure
DbImporter
DbExporter

[aiida.transport]
Transport
```

**Comment by ltalirz** [comment]
Some of the items you list are classes, some are modules.
Just for me to understand: when you write e.g. 
```
[aiida.cmdline]
params.arguments
```
this means: all functions and classes defined in `aiida.cmdline.params.arguments` are exposed at `aiida.cmdline`?

**Comment by sphuber** [comment]
Yes, when it is a module, that means anything that is exposed on the `__init__.py` level, defined by the `__all__`, will be exposed

**Comment by sphuber** [comment]
Fixed in PR #2528
